src/transformers/models/mobilevit/modeling_mobilevit.py

```diff
@@ -437,8 +438,16 @@ def unfolding(self, features: torch.Tensor) -> Tuple[torch.Tensor, Dict]:
 
         batch_size, channels, orig_height, orig_width = features.shape
 
-        new_height = int(math.ceil(orig_height / patch_height) * patch_height)
-        new_width = int(math.ceil(orig_width / patch_width) * patch_width)
+        new_height = (
+            safe_float(torch.ceil(orig_height / patch_height) * patch_height)
+            if torch.jit.is_tracing()
```

Comment:
Do we need the conditional here? This is already handled in the `safe_float` and `safe_int` functions

Reply:
I think it's required for `torch.ceil` no?

Reply:
tbh, I don't know, is `torch.ceil` not tracing compatible?

Reply:
one needs to convert a data type to a tensor before using `torch.ceil` AFAIK

Reply:
if I'm passing an int or float, torch.ceil will be called first and it will fail because torch.ceil can only be called with tensors AFAIK

Reply:
Makes sense!

Reply:
Only other Q here then is why do we use a `float` when tracing and int otherwise?

Reply:
sorry I think I was mistaken with that one, you're right, I fixed it :)